### DISCUSSION TITLE
How to add multiple text label on region

### DISCUSSION THREAD

**Discussion by abinasn** [katspaugh/wavesurfer.js#3173] [Q&A]
I was using wavesurfer v6. Currently I migrated from v6 to v7. I see in Region plugins `attributes` has been removed while creating region, i.e. `addRegion`. My v6 code snippet below.
```
ws.current.on("ready", () => {
  segments.forEach(seg => {
    ws.current.addRegion({
      start: seg.startTime,
      end: seg.endTime,
      data: seg,
      attributes: {
        label: seg.target,
        sourceLabel: seg.source
      }
    });
  })
});
```

How can I achieve the same in V7. can  anyone help?

**Comment by katspaugh** [comment]
Yes, the new parameters are [these](https://wavesurfer-js.org/docs/types/plugins_regions.RegionParams).
The closest one to a `label` would be `content`.

**Reply by rorr31** [reply]
I am also attempting to update from 6 to 7.  In addition to the label, we also used the attributes to store other custom metadata particular to our use case and functionality.  Is there any way in 7 to attach custom metadata to a region now that the attributes property is removed?

Also, is there a way to customize the styling of the content?  For instance, where it appears within the region and the font-size?

**Reply by katspaugh** [reply]
Content can be an HTML element which you can style on your own. Just keep in mind it's added to Shadow DOM so global classes won't work. You can use the `part` attribute tho, as shown here: https://wavesurfer.xyz/examples/?styling.js

Reagrding custom attributes, you can either add them directly to a region object, or have a side object that stores that data by region id.

E.g.

Attaching directly to a region:
```js
const region = regionsPlugin.addRegion({ ... })
region.customData = '...'
region.customAnything = { ... }
```

Or as a separate map:

```js
const allRegionsData = {}

const region = regionsPlugin.addRegion({
  id: 'my-region',
  // ...
})

allRegionsData[region.id] = customData
```

**Reply by rorr31** [reply]
Awesome.  I'll try this out.  Thank you!!!

**Reply by rorr31** [reply]
Another issue I'm running into with the migration from 6 to 7 (Please let me know if you'd like me to add this as an issue or submit somewhere else):

In my interface, I create regions and allow users to resize/drag them.  When an update of the region occurs, it can affect the size of neighboring regions.  In 6, I could use the isDragging and isResizing parameters to determine the action that was being taken and then get the start and end values to affect the size of the bordering regions.  

It looks like in 7, those methods are removed.  So, to determine whether dragging or resizing is occurring, I wrote an on 'update' function to examine the start and end values to determine what's happening.  However, the start and end values are always the same on every invocation of the method even though a resize is happening.

Here is the logic within the on update function:

```
const
	attributes = region.attributes,
	origStart = attributes.origStart,
	origEnd = attributes.origEnd,
	start = region.start,
	end = region.end;

console.log(start, origStart, end, origEnd);

if (start !== origStart || end !== origEnd) {

	if (end <= origEnd && start >= origStart) {
		console.log('Contracting');
		if (end - start < MIN_WORD_DURATION_SECONDS) {
			if (end < origEnd) {
				region.end = start + MIN_WORD_DURATION_SECONDS;
			} else {
				region.start = end - MIN_WORD_DURATION_SECONDS;
			}
		}
	} else if (end - origEnd === start - origStart) {
		console.log('Dragging');
		if (end > origEnd) {
			processDragRight(region);
		} else {
			processDragLeft(region);
		}
	} else {
		console.log('Resizing');
		if (end > origEnd) {
			processResizeRight(region);
		} else {
			processResizeLeft(region);
		}
	}
}
```

The problem is nothing ever fires because end always equals origEnd and start always equals origStart.  What am I missing?

**Reply by katspaugh** [reply]
Not sure what you're missing but if you just want to detect if a region has been resized vs moved, you can do it like this: https://wavesurfer.xyz/examples/?2a124b3f28a9320676aa6c300a563fae

I think it would make sense to extend the update event with a parameter to make this easier, so please feel free to make an issue for this.

**Comment by abinasn** [comment]
And also `rtl` support has been removed from wavesurfer create event. Only in css `part(region-content)` it can be added. 

Is there any way i can do it in Region plugin. I wanted to make it conditional based on arabic languages.
